### PR TITLE
Krig

### PR DESCRIPTION
**Pull request by stefraynaud**
Add kriging support

### CODE REVIEW

File: xoa/krig.py [hunk 1]
```diff
@@ -0,0 +1,867 @@
...
+        """Get the :attr:`estimated_params` as an array
+
+        Returns
+        -------
+        numpy.array
+        """
+        pp = list(self.estimated_params.values())
+        if None in pp:
+            raise VariogramModelError(
+                "No all parameters are estimated: {}".format(self.estimated_params)
```
Comment (vquilfen) [inline xoa/krig.py:395]: Not all

**Review by mdussa — APPROVED** [review]

File: xoa/krig.py [hunk 2]
```diff
@@ -0,0 +1,867 @@
...
+
+    def get_func(self, pp=None):
+        """Get the variogram model function using `pp` variable arguments"""
+        if pp is not None:
+            params = self.set_params_array(pp)
+        else:
+            params = self.get_params()
+            if None in list(params.values()):
+                raise VariogramModelError(
+                    "No all parameters are estimated: {}".format(self.estimated_params)
```
Comment (vquilfen) [inline xoa/krig.py:436]: Not all

File: xoa/krig.py [hunk 3]
```diff
@@ -0,0 +1,867 @@
...
+    def fit(self, da: xr.DataArray, **kwargs):
+        """Estimate parameters from data"""
+        # We need a data array
+        if isinstance(da, xr.Dataset):
+            if len(da) == 1:
+                da = da[list(da)[0]]
+            elif len(da) > 1:
+                xoa_warn(
+                    "Multiple candidate variable found in the dataset for estimating "
+                    "the variogram parameters. Keeping only the forst one."
```
Comment (vquilfen) [inline xoa/krig.py:449]: first

File: xoa/krig.py [hunk 4]
```diff
@@ -0,0 +1,867 @@
...
+        """Plot the semivariogram
+
+        Parameters
+        ----------
+        rmax: float
+            Max range in meters
+        nr: int
+            Number of points to plot the curve
+        show_params: bool, dict
+            Show a text box that contains the variogram paramters in the lower right corner.
```
Comment (vquilfen) [inline xoa/krig.py:507]: parameters

File: xoa/krig.py [hunk 5]
```diff
@@ -0,0 +1,867 @@
...
+    are split into smaller clusters using cluster analysis of distance with
+    function :func:`~xoa.geo.clusterize`.
+
+    The problem is solved in this way:
+
+        #. Input points are split in clusters if necessary.
+        #. The input variogram matrix is inverted
+            for each cluster, possibly using
+            :mod:`multiprocessing` if ``nproc>1``.
+        #. Value are computed at output positions
```
Comment (vquilfen) [inline xoa/krig.py:578]: Values

File: xoa/krig.py [hunk 6]
```diff
@@ -0,0 +1,867 @@
...
+            using each the inverted matrix of cluster.
```
Comment (vquilfen) [inline xoa/krig.py:579]: each using ?

**Review by vquilfen — APPROVED** [review]

**Pull request merged**